File: andriodGL/app/src/main/java/z3d/frame3d/FrameNodeVo.java
package z3d.frame3d;

import android.util.Log;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import z3d.display.BuildDisplay3DSprite;

public class FrameNodeVo {
    private static final String TAG = "FrameNodeVo";
    public int type;
    public int id;
    public String name;
    public String url;
    public String resurl;
    public float curTime=1;
    public boolean noLight;
    public boolean directLight;
    public boolean receiveShadow;
    public String lighturl;

    public List<FrameLinePointVo> pointitem;
    public List materialInfoArr;

    public String materialurl;


    public float    maxTime;

    public void writeObject(JSONObject $obj) {
        try {
        this.id = $obj.getInt("id");
        this.name = $obj.getString("name");
        this.url = $obj.getString("url");
        JSONArray itememArr= $obj.getJSONArray("pointitem");
        this.pointitem = new ArrayList<>();
        for (int j = 0; j < itememArr.length(); j++) {
            FrameLinePointVo frameLinePointVo = new FrameLinePointVo();
            frameLinePointVo.writeObject((JSONObject) itememArr.get(j));
            this.maxTime=Math.max(frameLinePointVo.time,this.maxTime);
            this.pointitem.add(frameLinePointVo);
        }
        this.resurl = $obj.getString("resurl");
            if (   this.url.indexOf(".prefab")!=-1) {
                this.materialInfoArr = new ArrayList();
                JSONArray materialInfoArr=$obj.getJSONArray("materialInfoArr");
                for (int i = 0;  i < materialInfoArr.length(); i++) {
                    this.materialInfoArr.add(materialInfoArr.get(i));
                }
                this.noLight = $obj.getBoolean("noLight");
                this.directLight =$obj.getBoolean("directLight");
                this.receiveShadow =$obj.getBoolean("receiveShadow");
                if (this.noLight == false) {
                    this.lighturl = $obj.getString("lighturl");
                }
                this.materialurl = $obj.getString("materialurl");
                this.type = 1;
            }
            if (   this.url.indexOf(".lyf")!=-1) {
                this.type = 2;
            }
            if (   this.url.indexOf(".zzw")!=-1) {
                this.type = 3;
            }

        } catch (Exception e) {
            e.printStackTrace();
        }




    }

}

File: h5gl/src/Pan3d/frame3d/FrameNodeVo.ts
module Pan3d {

    export class FrameNodeVo {
        constructor() {
        }
        public type: number;
        public id: number;
        public name: number;
        public url: string;
        public resurl: string;

        public noLight: boolean;
        public directLight: boolean;
        public receiveShadow: boolean;
        public lighturl: string

        public pointitem: Array<FrameLinePointVo>;
        public materialInfoArr: Array<any>;

        public materialurl: string

        public writeObject($obj: any): void {

            this.id = $obj.id;
            this.name = $obj.name;
            this.url = $obj.url;


            this.pointitem = new Array
            for (var j: number = 0; j < $obj.pointitem.length; j++) {
                var $FrameLinePointVo: FrameLinePointVo = new FrameLinePointVo();
                $FrameLinePointVo.writeObject($obj.pointitem[j])
                this.pointitem.push($FrameLinePointVo)
            }
            this.resurl = $obj.resurl
            if (this.url.search(".prefab") != -1) {
                this.materialInfoArr = new Array
                for (var i: number = 0; $obj.materialInfoArr && i < $obj.materialInfoArr.length; i++) {
                    this.materialInfoArr.push($obj.materialInfoArr[i])
                }
                this.noLight = $obj.noLight;
                this.directLight = $obj.directLight;
                this.receiveShadow = $obj.receiveShadow;

                if (this.noLight == false) {
                    this.lighturl = $obj.lighturl

                }
                this.materialurl = $obj.materialurl
                this.type = 1;
            }
            if (this.url.search(".lyf") != -1) {
                this.type = 2;
            }
            if (this.url.search(".zzw") != -1) {
                this.type = 3;
            }



        }



    }
}